scrypt: 1.3.0 → 1.3.1, build library, enable tests

sternenseemann opened the pull request:
* update scrypt
* enable running of tests
* build development library libscrypt-kdf,
  install to lib output, headers to dev
* default output remains untouched: contains binary plus man pages

cc @thoughtpolice 

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

sternenseemann committed:
scrypt: 1.3.0 → 1.3.1, build library, enable tests

* update scrypt
* enable running of tests
* build development library libscrypt-kdf,
  install to lib output, headers to dev
* default output remains untouched: contains binary plus man pages

sternenseemann commented:
Odd thing about this, is that it doesn't install the `crypto/crypto_scrypt.h` header like [the documentation says](https://www.tarsnap.com/scrypt.html), but debian also has it this way, not sure what we should do.

Also: Do we install `*.a` to `dev` or `lib`?

thoughtpolice commented:
`.a` files should go in `$lib`, while `$dev` should contain header files only. You can also add a `$man` output so that `$out` won't needlessly contain the man pages too. I wouldn't block it on that, but it would be a nice to have.

The lack of `crypto_scrypt.h` is perhaps more concerning; it's also possible this is a bug in Debian's installer or even the scrypt package itself. What do the consumers of `libscrypt-kdf` typically ask for?

sternenseemann commented:
> The lack of crypto_scrypt.h is perhaps more concerning; it's also possible this is a bug in Debian's installer or even the scrypt package itself. What do the consumers of libscrypt-kdf typically ask for?

Seems like it was a conscious decision to rename the header and main function for the shared library: <https://github.com/Tarsnap/scrypt/pull/168#issuecomment-485275045> I also found one instance of `libscrypt-kdf` being used in actual C code with a quick github search and there the [`scrypt-kdf.h` header is used](https://github.com/rmind/rvault/blob/master/src/crypto/kdf.c#L24).

I'll open an issue about the misleading documentation on their website.


> .a files should go in $lib, while $dev should contain header files only.

Then this is already working correctly.

> You can also add a $man output so that $out won't needlessly contain the man pages too. I wouldn't block this PR on that or anything, but it would be a nice to have.

I've generally experienced hesitance to add a new output for man pages of small size (or a single one in this case), so I didn't add a separate output (seemingly the consensus in nixpkgs seems to be to use them sparingly?), but I'd be happy to add one if you prefer it.

sternenseemann committed:
scrypt: fix build of pkgsStatic.scrypt

The getconf input defaults to the glibc one if it is being used and uses
the netbsd version in all other cases. This fixes the build when
building with musl, since it doesn't ship a version of getconf.

sternenseemann commented:
@GrahamcOfBorg build pkgsStatic.scrypt

gperciva commented:
@sternenseemann has correctly summarized scrypt's position about `scrypt_kdf()`: that's the intended public API for `libscrypt-kdf`.  `crypto_scrypt()` was the function to use before we had a library, when people just copied our files into their projects directly.

We'll be updating our `README.md` and website soon.
